STIPS/STIP-009.md

```diff
@@ -0,0 +1,1178 @@
+```
+
++ returns allowedAssets array
+
+----
+
+> isInitializedExtension
+
+```solidity
+function isInitializedExtension(address _extension)
+```
+
++ if _extension state in *extensionAllowList* is INITIALIZED, return true. Otherwise false;
+
+----
+
+> isPendingExtension
+
+```solidity
+function isPendingExtension(address _extension)
+```
+
++ if _extension state in *extensionAllowList* is PENDING, return true. Otherwise false;
+
+> isAllowedAsset
+
+```solidity
+function isAllowedAsset(address _asset)
+```
+
++ return allowedAssetList(_asset)
+
+
+### BaseGlobalExtension
+
+#### Modifiers
+
+> onlyAssetAllowList
+
+```solidity
+modifier onlyAssetAllowList(address memory _receiveAsset) {
+    if (manager.useAllowList) {
+        require(manager.assetAllowList[_receiveAsset], "Must be allowed asset");
+    }
+    _;
+}
+```
+
+> onlyOwner
+
+```solidity
+modifier onlyOwner(ISetToken _setToken) {
+    require(msg.sender == _manager(_setToken).owner(), "Must be owner");
+    _;
+}
+```
+
+> onlyMethodologist
+
+```solidity
+modifier onlyMethodologist(ISetToken _setToken) {
+    require(msg.sender == _manager(_setToken).methodologist(), "Must be methodologist");
+    _;
+}
+```
+
+> onlyOperator
+
+```solidity
+modifier onlyOperator(ISetToken _setToken) {
+    require(_manager(_setToken).operatorAllowlist(msg.sender), "Must be approved operator");
+    _;
+}
+```
+
+> onlyManager
+
+```solidity
+modifier onlyManager(ISetToken _setToken) {
+    require(address(_manager(_setToken)) == msg.sender, "Manager must be sender");
+    _;
+}
+```
+
+### Internal functions
+
+> invokeManager
+
+Invoke call from manager
+
+```solidity
+function invokeManager(ISetToken _setToken, address _module, bytes memory _encoded)
+```
+
++ call *_manager(_setToken).interactManager(_module, _encoded)*
+
+### Abstract Functions
+
+> _manager
+
+Internal function to grab manager of passed SetToken from extensions data structure
+
+```solidity
+function _manager(ISetToken _setToken) internal virtual view returns (IDelegatedManager)
+```
+
+----
+
+> removeExtension
+
+Removes extension
+
+```solidity
+function removeExtension(ISetToken _setToken) external virtual;
+```
+
+### TradeExtension
+
+#### Inheritance
+
+- BaseGlobalExtension
+
+#### Global Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|ITradeModule|tradeModule|Trade Module for Set Token|
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|mapping(address => IDelegatedManager)|setManagers|Mapping from Set Token to DelegatedManager|
+
+#### Functions
+
+| Name  | Caller  | Description     |
+|------	|------	|-------------	|
+|initialize|owner|Initialize the TradeExtension on the DelegatedManager and initialize the TradeModule on the SetToken if necessary|
+|trade|operator|Trade between whitelisted assets on a DEX|
+
+----
+
+### Functions
+
+> initializeExtension
+
+ONLY OWNER: Initialize the TradeExtension on the DelegatedManager
```

Comment:
Just FYI we can't use the `onlyOwner` modifier here since it's reliant on `_manager` which won't be filled out until after the extension has been initialized